src/components/dictionaryConcepts/components/CreateMapping.jsx

```diff
@@ -31,7 +31,7 @@ class CreateMapping extends Component {
             className="form-control"
             name="source"
             onChange={(event) => { updateEventListener(event, url); }}
-            defaultValue={isNew ? '' : source}
+            value={source || undefined}
           >
             <option value="" hidden>Select a source</option>
```

Comment:
When the value property is set to an empty string, it will select the option that says `"Select a source"` because their values will match. However, if it is set to `undefined` it will adopt the value selected by the user from the dropdown (when Creating) and the correct source value (when editing).

Reply:
Ok